metals/src/main/scala/scala/meta/internal/metals/debug/DebugDiscovery.scala

```diff
@@ -162,9 +172,226 @@ class DebugDiscovery(
+      }
+
+      val syntheticOccOpt = if (occOpt.isEmpty) {
+        textDocument.symbols
+          .find(_.symbol.contains(mainClass.getClassName.split('.').last))
+          .flatMap(symbolInfo =>
+            textDocument.occurrences.find(_.symbol == symbolInfo.symbol)
+          )
+      } else None
+
+      (occOpt.orElse(syntheticOccOpt)).flatMap(_.range).map { range =>
+        val lineDiff = math.abs(range.startLine - position.getLine)
+        val charDiff = math.abs(range.startCharacter - position.getCharacter)
+        (lineDiff, charDiff, "main", mainClass.getClassName, mainClass)
+      }
+    }
+  }
+
+  private def calculateTestCaseDistances(
+      allTestCases: List[(String, List[TestCaseEntry])],
+      position: Position,
+  ): Seq[(Int, Int, String, String, String)] = {
+    allTestCases.flatMap { case (suiteFqn, cases) =>
+      cases.map { entry =>
+        val range = entry.location.getRange()
+        val start = range.getStart()
+        val lineDiff = math.abs(start.getLine() - position.getLine)
+        val charDiff = math.abs(start.getCharacter() - position.getCharacter)
+        (lineDiff, charDiff, "testCase", entry.name, suiteFqn)
+      }
+    }
+  }
+
+  private def calculateTestSuiteDistances(
+      textDocument: TextDocument,
+      allTestCases: List[(String, List[TestCaseEntry])],
+      position: Position,
+  ): Seq[(Int, Int, String, String, String)] = {
+    val testClassOccurrences = textDocument.occurrences.filter { occ =>
+      occ.role.isDefinition && (
+        allTestCases.exists { case (suiteFqn, _) =>
+          occ.symbol.contains(suiteFqn.split('.').last)
+        }
+      )
+    }
+
+    testClassOccurrences.flatMap { occ =>
+      occ.range.flatMap { range =>
+        val lineDiff = math.abs(range.startLine - position.getLine)
+        val charDiff = math.abs(range.startCharacter - position.getCharacter)
+
+        allTestCases.headOption
+          .map(_._1)
+          .map { suiteFqn =>
+            (lineDiff, charDiff, "testSuite", "", suiteFqn)
+          }
+      }
+    }
+  }
+
+  private def findClosestTarget(
+      allDistances: Seq[(Int, Int, String, String, Any)]
+  ): Option[(Int, Int, String, String, Any)] = {
+    allDistances.sortBy { case (lineDiff, charDiff, targetType, _, _) =>
+      val typePriority = targetType match {
+        case "main" => 0
+        case "testCase" => 1
+        case "testSuite" => 2
+        case _ => 3
+      }
+      (lineDiff, charDiff, typePriority)
+    }.headOption
+  }
+
+  private def createTestSuiteParams(
+      buildTarget: b.BuildTargetIdentifier,
+      suiteFqn: String,
+  ): Future[b.DebugSessionParams] = {
+    import scala.jdk.CollectionConverters._
+    val params = new b.DebugSessionParams(singletonList(buildTarget))
+    params.setDataKind(b.TestParamsDataKind.SCALA_TEST_SUITES_SELECTION)
+    val suiteSelection =
+      new b.ScalaTestSuiteSelection(suiteFqn, List.empty[String].asJava)
+    val suites = new b.ScalaTestSuites(
```

Comment:
ScalaTestSuite is contructed also elsewhere in file, it might be worth to extract that logic somewhere especially that it includes additional jvm options there

This is also very similar to the below method, so might be worth to deduplicate that code